client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -599,8 +599,15 @@ public V call() throws Exception {
       return new MetadataValueImpl<V>(value.getCreated(), value.getLifespan(), value.getLastUsed(), value.getMaxIdle(), value.getVersion(), valueObj);
    }
 
-   private int toSeconds(long duration, TimeUnit timeUnit) {
-      return (int) timeUnit.toSeconds(duration);
+   protected static int toSeconds(long duration, TimeUnit timeUnit) {
+      int seconds = (int) timeUnit.toSeconds(duration);
+      long inverseDuration = timeUnit.convert(seconds, TimeUnit.SECONDS);
+
+      if (duration > inverseDuration) {
+         //Round up.
+         seconds++;
```

Comment:
I think this can be intergrated as it is as it solves the two issues mentioned in the description. I'm attempting to properly resolve the issue of having entries still alive for less than 1sec after expiry in https://issues.jboss.org/browse/ISPN-5298